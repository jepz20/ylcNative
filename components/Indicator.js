// @flow

import * as React from 'react'
import { View, StyleSheet } from 'react-native'

type Props = {
  type: 'win' | 'loose' | 'tie' | 'neutral'
}
const Indicator: React.StatelessFunctionalComponent<Props> = ({
  type = 'neutral'
}) => {
  return <View style={[styles.indicator, styles[type]]} />
}

const styles = StyleSheet.create({
  indicator: {
    borderRadius: 50,
    height: 30,
    width: 30,
    borderWidth: 1,
    elevation: 2,
    borderColor: '#000'
  },
  win: {
    backgroundColor: 'green'
  },
  loose: {
    backgroundColor: 'red'
  },
  tie: {
    backgroundColor: 'yellow'
  },
  neutral: {
    backgroundColor: 'grey'
  }
})

export { Indicator }
