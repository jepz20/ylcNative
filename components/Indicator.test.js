import { Text } from 'react-native'
import React from 'react'
import { Indicator } from './Indicator'

import renderer from 'react-test-renderer'

describe('Indicator', () => {
  test('renders with type win ', () => {
    const indicator = renderer.create(<Indicator type='win' />).toJSON()
    expect(indicator).toMatchSnapshot()
  })

  test('renders with type loose ', () => {
    const indicator = renderer.create(<Indicator type='loose' />).toJSON()
    expect(indicator).toMatchSnapshot()
  })

  test('renders with type tie ', () => {
    const indicator = renderer.create(<Indicator type='loose' />).toJSON()
    expect(indicator).toMatchSnapshot()
  })

  test('renders with type neutral', () => {
    const indicator = renderer
      .create(<Indicator>Shouldn't render</Indicator>)
      .toJSON()
    expect(indicator).toMatchSnapshot()
  })

  test('renders with default props as neutral', () => {
    const indicator = renderer
      .create(<Indicator>Shouldn't render</Indicator>)
      .toJSON()
    expect(indicator).toMatchSnapshot()
  })
})
