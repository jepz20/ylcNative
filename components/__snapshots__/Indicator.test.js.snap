// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indicator renders with default props as neutral 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#000",
        "borderRadius": 50,
        "borderWidth": 1,
        "elevation": 2,
        "height": 30,
        "width": 30,
      },
      Object {
        "backgroundColor": "grey",
      },
    ]
  }
/>
`;

exports[`Indicator renders with type loose  1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#000",
        "borderRadius": 50,
        "borderWidth": 1,
        "elevation": 2,
        "height": 30,
        "width": 30,
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
/>
`;

exports[`Indicator renders with type neutral 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#000",
        "borderRadius": 50,
        "borderWidth": 1,
        "elevation": 2,
        "height": 30,
        "width": 30,
      },
      Object {
        "backgroundColor": "grey",
      },
    ]
  }
/>
`;

exports[`Indicator renders with type tie  1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#000",
        "borderRadius": 50,
        "borderWidth": 1,
        "elevation": 2,
        "height": 30,
        "width": 30,
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
/>
`;

exports[`Indicator renders with type win  1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#000",
        "borderRadius": 50,
        "borderWidth": 1,
        "elevation": 2,
        "height": 30,
        "width": 30,
      },
      Object {
        "backgroundColor": "green",
      },
    ]
  }
/>
`;
